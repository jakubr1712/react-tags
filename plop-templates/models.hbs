export interface I{{properCase name}}ComponentProps {}