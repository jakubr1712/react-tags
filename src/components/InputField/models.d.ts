export interface IInputFieldProps {
  input: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onClear: () => void;
  onFocus: () => void;
}
