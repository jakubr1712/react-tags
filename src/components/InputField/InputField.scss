.input-field {
  position: relative;
  width: 100%;

  &__input {
    padding: 8px 32px 8px 32px;
    width: 100%;
    box-sizing: border-box;
    margin-bottom: 5px;
    border: none;

    &:focus {
      outline: none;
      border: none;
    }
  }

  &__search-icon {
    position: absolute;
    left: 8px;
    top: 28%;
    pointer-events: none;
  }

  &__clear-icon {
    position: absolute;
    right: 8px;
    top: 20%;
    cursor: pointer;
    color: #999;
  }
}
