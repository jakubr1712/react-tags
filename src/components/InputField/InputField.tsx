import React from "react";
import "./InputField.scss";
import { IInputFieldProps } from "./models.d";

const InputField: React.FC<IInputFieldProps> = ({
  input,
  onChange,
  onClear,
  onFocus,
}) => {
  return (
    <div className="input-field">
      <span className="input-field__search-icon">
        <svg
          data-testid="search-icon"
          viewBox="0 0 24 24"
          width="16"
          height="16"
          fill="black"
        >
          <path d="M10 2C5.58 2 2 5.58 2 10s3.58 8 8 8c1.64 0 3.16-.53 4.41-1.41L21.59 22 23 20.59l-7.18-7.18C17.47 13.16 18 11.64 18 10c0-4.42-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z" />
        </svg>
      </span>
      <input
        className="input-field__input"
        value={input}
        onChange={onChange}
        placeholder="Wyszukaj grupÄ™ lub tag"
        onFocus={onFocus}
      />
      <span className="input-field__clear-icon" onClick={onClear}>
        &times;
      </span>
    </div>
  );
};

export default InputField;
