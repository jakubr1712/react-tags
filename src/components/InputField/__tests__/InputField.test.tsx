import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import InputField from "../InputField";
import { IInputFieldProps } from "../models.d";

describe("InputField component", () => {
  const defaultProps: IInputFieldProps = {
    input: "",
    onChange: jest.fn(),
    onClear: jest.fn(),
    onFocus: jest.fn(),
  };

  it("renders properly and matches snapshot", () => {
    const { container } = render(<InputField {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the search icon", () => {
    render(<InputField {...defaultProps} />);
    expect(screen.getByTestId("search-icon")).toBeInTheDocument();
  });

  it("renders the input field with the correct placeholder", () => {
    render(<InputField {...defaultProps} />);
    expect(
      screen.getByPlaceholderText("Wyszukaj grupę lub tag")
    ).toBeInTheDocument();
  });

  it("calls onChange when input value changes", () => {
    render(<InputField {...defaultProps} />);
    fireEvent.change(screen.getByPlaceholderText("Wyszukaj grupę lub tag"), {
      target: { value: "test" },
    });
    expect(defaultProps.onChange).toHaveBeenCalled();
  });

  it("calls onFocus when the input field is focused", () => {
    render(<InputField {...defaultProps} />);
    fireEvent.focus(screen.getByPlaceholderText("Wyszukaj grupę lub tag"));
    expect(defaultProps.onFocus).toHaveBeenCalled();
  });

  it("calls onClear when the clear icon is clicked", () => {
    render(<InputField {...defaultProps} />);
    fireEvent.click(screen.getByText("×"));
    expect(defaultProps.onClear).toHaveBeenCalled();
  });
});
