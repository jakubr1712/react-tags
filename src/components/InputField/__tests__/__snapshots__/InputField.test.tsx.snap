// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component renders properly and matches snapshot 1`] = `
<div>
  <div
    class="input-field"
  >
    <span
      class="input-field__search-icon"
    >
      <svg
        data-testid="search-icon"
        fill="black"
        height="16"
        viewBox="0 0 24 24"
        width="16"
      >
        <path
          d="M10 2C5.58 2 2 5.58 2 10s3.58 8 8 8c1.64 0 3.16-.53 4.41-1.41L21.59 22 23 20.59l-7.18-7.18C17.47 13.16 18 11.64 18 10c0-4.42-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z"
        />
      </svg>
    </span>
    <input
      class="input-field__input"
      placeholder="Wyszukaj grupę lub tag"
      value=""
    />
    <span
      class="input-field__clear-icon"
    >
      ×
    </span>
  </div>
</div>
`;
